sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -159,6 +161,26 @@ object RapidsPluginUtils extends Logging {
         logWarning(s"$numThreadsKey is set to $numThreads.")
       }
     }
+
+    // If spark.task.resource.gpu.amount is larger than 
+    // (spark.executor.resource.gpu.amount / spark.executor.cores) then GPUs will be the limiting
+    // resource for task scheduling.
+    if (conf.contains(TASK_GPU_AMOUNT_KEY) && conf.contains(EXECUTOR_GPU_AMOUNT_KEY)) {
+      val taskGpuAmountSetByUser = conf.get(TASK_GPU_AMOUNT_KEY).toDouble
+      // get worker's all cores num if spark.executor.cores is not set explicitly
+      val executorCores = if (conf.contains(EXECUTOR_CORES_KEY)) {
+        conf.get(EXECUTOR_CORES_KEY).toDouble
+      } else {
+        Runtime.getRuntime.availableProcessors.toDouble
+      }
```

Comment:
Minor Nit: This does an unnecessary double-lookup (contains followed by get):
```suggestion
      val executorCores = conf.getOption(EXECUTOR_CORES_KEY)
          .map(_.toDouble)
          .getOrElse(Runtime.getRuntime.availableProcessors.toDouble)
```